wallet/core/src/tests/rpc_core_mock.rs

```diff
@@ -0,0 +1,258 @@
+use crate::imports::*;
+
+use async_channel::{unbounded, Receiver};
+use async_trait::async_trait;
+use kaspa_notify::events::EVENT_TYPE_ARRAY;
+use kaspa_notify::listener::ListenerId;
+use kaspa_notify::notifier::{Notifier, Notify};
+use kaspa_notify::scope::Scope;
+use kaspa_rpc_core::api::ctl::RpcCtl;
+use kaspa_rpc_core::{api::rpc::RpcApi, *};
+use kaspa_rpc_core::{notify::connection::ChannelConnection, RpcResult};
+use std::sync::Arc;
+
+pub type RpcCoreNotifier = Notifier<Notification, ChannelConnection>;
+
+impl From<Arc<RpcCoreMock>> for Rpc {
+    fn from(rpc_mock: Arc<RpcCoreMock>) -> Self {
+        Self::new(rpc_mock.clone(), rpc_mock.ctl.clone())
+    }
+}
+
+pub struct RpcCoreMock {
+    ctl: RpcCtl,
+    core_notifier: Arc<RpcCoreNotifier>,
+    _sync_receiver: Receiver<()>,
+}
```

Comment:
Many thanks, much appreciated!  

Re code: these are just some initial structs on my end to try to simulate node processing so that we can construct in-depth integration tests against the wallet API, but I need to discuss this later with @tiram88 to see if we can use a local simnet/devnet to build this as the node already has all this functionality for internal integration tests but I am not familiar with it. Perhaps @coderofstuff can help as well.  These are just initial efforts to see if some preliminary processing can be done within unit tests for early traps (this is just scaffolding for now, but I am rolling it in to see how this can be adopted/evolved down the road).